crates/hir-expand/src/eager.rs

```diff
@@ -60,29 +72,25 @@ pub fn expand_eager_macro(
         def,
         krate,
         eager: Some(Box::new(EagerCallInfo {
-            arg_or_expansion: Arc::new(parsed_args.clone()),
+            arg_or_expansion: Arc::new(parsed_args),
             included_file: None,
             error: None,
         })),
         kind: MacroCallKind::FnLike { ast_id: call_id, expand_to: ExpandTo::Expr },
     });
 
-    let parsed_args = mbe::token_tree_to_syntax_node(&parsed_args, mbe::TopEntryPoint::Expr).0;
```

Comment:
the actual fix was just this part, we turned the tokentree node into a subtree and then back into a syntax node. All this does is effectively discarding whitespace which ruins the tokenmap calculated by the maro_arg query (that is not used itself for eager expansion, but is for mapping tokens)

Reply:
Ah no I misunderstood actually, we transform the token tree to an expression. Okay then macro_arg needs changes